core/domain/email_manager.py

```diff
@@ -255,3 +285,70 @@ def send_moderator_action_email(
     _send_email(
         recipient_id, sender_id, intent, email_subject, full_email_content,
         bcc_admin=True)
+
+
+def send_role_notification_email(
+        inviter_id, recipient_id, recipient_role, exploration_id,
+        exploration_title):
+    """Sends a email when a new user is given activity rights (Manager, Editor,
+    Viewer) to an exploration by creator of exploration.
+
+    Email will only be sent if recipient wants to receive these emails (i.e.
+    'can_receive_editor_role_email' is set True in recipent's preferences).
+    """
+    # Editor role email body and email subject templates.
+    # pylint: disable=invalid-name
+    editor_role_email_subject_template = (
+        '%s invited you to collaborate on Oppia.org')
+
+    editor_role_email_body_template = (
+        '%s has granted you %s to their learning exploration, '
+        '<a href="http://www.oppia.org/%s">%s</a>'
+        ', on Oppia.org.<br>'
+        'This allows you:<br>'
+        '<ul>%s</ul><br>'
+        'You can find exploration'
+        '<a href="http://www.oppia.org/%s">here</a>. Thanks!<br>'
+        '<br>Best Wishes,<br>'
+        'The Oppia Team<br>'
+    )
+
+    # Return from here if sending email is turned off.
+    if not feconf.CAN_SEND_EMAILS_TO_USERS:
+        log_new_error('This app cannot send emails to users.')
+        return
+
+    # Return from here is sending editor role email is disabled.
+    if not feconf.CAN_SEND_EDITOR_ROLE_EMAILS:
+        log_new_error('This app cannot send editor role emails to users.')
+        return
+
+    recipient_user_settings = user_services.get_user_settings(recipient_id)
+    inviter_user_settings = user_services.get_user_settings(inviter_id)
+    recipient_preferences = user_services.get_email_preferences(recipient_id)
+
+    if not recipient_preferences['can_receive_editor_role_email']:
+        # Do not send email if recipient has declined.
+        return
+
+    if recipient_role not in EDITOR_ROLE_EMAIL_ROLES.keys():
+        raise Exception(
+            'Invalid role: %s' % recipient_role)
+
+    access_right_string = EDITOR_ROLE_EMAIL_ROLES[recipient_role]
+    rights_html = EDITOR_ROLE_EMAIL_RIGHTS_FOR_ROLE[access_right_string]
+
+    raw_email_body = 'Hi %s,<br><br>%s<br><br>%s' % (
```

Comment:
This is weird. Probably better to combine it with the email-body template above, and interpolate just a single string once and for all.